kore/src/Kore/Reachability/Claim.hs

```diff
@@ -609,10 +609,13 @@ simplify' lensClaimPattern claim = do
     simplifyRightHandSide lensClaimPattern sideCondition claim''
   where
     applySubstOnRightHandSide claimPat =
-        let substitution = Pattern.substitution $ Lens.view (field @"left") claimPat
-            noLeftSubst = Lens.set (field @"left" . field @"substitution") mempty claimPat
-            appliedSubst = ClaimPattern.applySubstitution substitution noLeftSubst
-         in appliedSubst
+        let substitution =
+                Pattern.substitution $ Lens.view (field @"left") claimPat
+            mapSubstitution ::
+                OrPattern RewritingVariableName -> OrPattern RewritingVariableName
+            mapSubstitution =
+                MultiOr.map (Lens.over (field @"substitution") (substitution <>))
+         in Lens.over (field @"right") mapSubstitution claimPat
```

Comment:
Some of these lens ops are not necessary and should be replaced with record syntax:
```haskell
applySubstOnRightHandSide :: ClaimPattern -> ClaimPattern
applySubstOnRightHandSide claimPat@ClaimPattern{left, right} =
  let substitution = Pattern.substitution left
      mapSubsitution ::
        Pattern RewritingVariableName ->
        Pattern RewritingVariableName
      mapSubstitution = Lens.over (field @"substitution") (substitution <>)
      right' = MultiOr.map mapSubstitution right
  in claimPat{right = right'}
```